services/web/server/src/simcore_service_webserver/projects/projects_api.py

```diff
@@ -916,7 +917,27 @@ async def patch_project_node(
     if not _user_project_access_rights.write:
         raise ProjectInvalidRightsError(user_id=user_id, project_uuid=project_id)
 
-    # 2. Patch the project node
+    # 2. If patching service key or version make sure it's valid
+    if _node_patch_exclude_unset.get("key") or _node_patch_exclude_unset.get("version"):
+        _project, _ = await db.get_project(
+            user_id=user_id, project_uuid=f"{project_id}"
+        )
+        _project_node_data = _project["workbench"][f"{node_id}"]
+
+        _service_key = _node_patch_exclude_unset.get("key", _project_node_data["key"])
+        _service_version = _node_patch_exclude_unset.get(
+            "version", _project_node_data["version"]
+        )
+        rabbitmq_rpc_client = get_rabbitmq_rpc_client(app)
+        await catalog_rpc.get_service(
```

Comment:
I can add a check function. `get_service`  is pretty expensive just to check that a service exists and the user has rights to it.